examples/custom-service.mdx

```diff
@@ -71,20 +157,37 @@ async fn init(
     Ok(PoolService { pool })
 }
 
-impl PoolService {
-    async fn start(&self) -> Result<(), CustomError> {
-        let (rec,): (String,) = sqlx::query_as("SELECT 'Hello world'")
-            .fetch_one(&self.pool)
-            .await
-            .map_err(CustomError::new)?;
-
-        assert_eq!(rec, "Hello world");
+impl CustomService {
+    async fn set_up_bot(&self) -> Result<(), Box<dyn std::error::Error>> {
+        let framework = poise::Framework::builder()
+            .options(poise::FrameworkOptions {
+                commands: vec![age()],
+                ..Default::default()
+            })
+            .token(&self.discord_api_key)
+            .intents(serenity::GatewayIntents::non_privileged())
+            .setup(|ctx, _ready, framework| {
+                Box::pin(async move {
+                    poise::builtins::register_globally(ctx, &framework.options().commands).await?;
+                    Ok(Data {})
+                })
+            });
+
+        framework.run().await.unwrap();
 
         Ok(())
     }
+
+    async fn set_up_router(&self, addr: std::net::SocketAddr) -> Result<(), hyper::Error> {
+        let router = router();
+
+        let router = axum::Server::bind(&addr).serve(router.into_make_service());
+
+        router.await
+    }
```

Comment:
This is good, but I think to be in line with how we do it for other services, we should configure the framework and the router in the `shuttle_service::main` function, and then start them both in the `Service` impl.

Reply:
I managed to refactor the framework setup back into the main function but it looks like trying to pass a router into the custom service struct causes the the compiler to complain about Axum's Router type not implementing Sync

Reply:
Ah, I think we'd need to use [syncwrapper](https://docs.rs/sync_wrapper/latest/sync_wrapper/struct.SyncWrapper.html) like we do for the axum service implementation.

Reply:
Ran into some issues at the start but it looks like I've got it working - will send a new revision with revised changes soonish since I've got it working on my example repo